### PR TITLE
Add config option for disabling Force Pushing

### PR DESCRIPTION
**Pull request by fsmiamoto**
This PR adds a new option for disabling the use of force push.

Since I'm new to the project, I'm not sure if there's a way to add some test related for it, but I think that would be ideal.

Any feedback or suggestion would be awesome :+1:

### CODE REVIEW

**Comment by fsmiamoto** [comment]
Related to #855

File: pkg/gui/files_panel.go [hunk 1]
```diff
@@ -517,14 +517,18 @@ func (gui *Gui) pushWithForceFlag(v *gocui.View, force bool, upstream string, ar
 		branchName := gui.getCheckedOutBranch().Name
 		err := gui.GitCommand.Push(branchName, force, upstream, args, gui.promptUserForCredential)
 		if err != nil && !force && strings.Contains(err.Error(), "Updates were rejected") {
+			forcePushDisabled := gui.Config.GetUserConfig().Get("git.disableForcePushing").(bool)
```
Comment (jesseduffield) [inline pkg/gui/files_panel.go:520]: You can go `.GetBool()` rather of .Get().(bool)

**Review by jesseduffield — CHANGES_REQUESTED** [review]
Congrats on making your first PR! Only the one comment but otherwise LGTM.

As for tests, we don't really have a proper test setup for the frontendy parts of the app. I intend on getting that setup when I switch from termbox to tcell, because that dependency has good support for testing.

**Comment by fsmiamoto** [comment]
Thanks for the review!
I've made the requested changes.

> I intend on getting that setup when I switch from termbox to tcell, because that dependency has good support for testing.

Nice, that would be great IMO 🙂

**Comment by jesseduffield** [comment]
LGTM, merging

**Pull request merged**